File: Greedy/2022. Convert 1D Array Into 2D Array/convert1DArrayInto2DArray.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        length = len(original)
        output = [[0 for _ in range(n)] for _ in range(m)]
        
        if m * n != length:
            return []
        
        for i in range(m):
            for j in range(n):
                # key is here:
                # row increasing i * col n + col increasing j
                output[i][j] = original[i * n + j]
        
        return output
        

File: Greedy/2022. Convert 1D Array Into 2D Array/convert1DArrayInto2DArray.js
/**
 * @param {number[]} original
 * @param {number} m
 * @param {number} n
 * @return {number[][]}
 */
 var construct2DArray = function(original, m, n) {
    let output = [];
    
    if (m * n !== original.length)
        return [];
    
    for (let i = 0; i < m; i++) {
        output.push([]);
        for (let j = 0; j < n; j++) {
            output[i][j] = original[i * n + j];
        }
    }
    
    return output;
};